### PR TITLE
5.x: fix more warnings in tests

### PR DESCRIPTION
**Pull request by LordSimal**

### CODE REVIEW

File: tests/TestCase/Utility/XmlTest.php [hunk 1]
```diff
@@ -573,7 +573,6 @@ public static function invalidArrayDataProvider(): array
             [[]],
             [['numeric key as root']],
             [['item1' => '', 'item2' => '']],
-            [['items' => ['item1', 'item2']]],
```
Comment (LordSimal) [inline tests/TestCase/Utility/XmlTest.php]: This caused [the following warning](https://github.com/cakephp/cakephp/actions/runs/12224365265/job/34097019729#step:13:259)

Comment (ADmad) [reply tests/TestCase/Utility/XmlTest.php]: The point of this provider is to test with invalid values, so this shouldn't be removed.

**Comment by ADmad** [comment]
Nice, these 2 are the only remaining now. Do you want to address them or punt it for later?

```
PHP Warning:  SimpleXMLElement::__construct(): Entity: line 3: parser error : Extra content at the end of the document in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298
PHP Warning:  SimpleXMLElement::__construct(): <items>item2</items> in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298
PHP Warning:  SimpleXMLElement::__construct(): ^ in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298

Warning: SimpleXMLElement::__construct(): Entity: line 3: parser error : Extra content at the end of the document in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298

Warning: SimpleXMLElement::__construct(): <items>item2</items> in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298

Warning: SimpleXMLElement::__construct(): ^ in /home/runner/work/cakephp/cakephp/src/Utility/Xml.php on line 298
```

```
PHP Warning:  fopen(/tmp/http_factory_tests_XHKeQd): Failed to open stream: `' is not a valid mode for fopen in /home/runner/work/cakephp/cakephp/vendor/laminas/laminas-diactoros/src/Stream.php on line 330
....
Warning: fopen(/tmp/http_factory_tests_XHKeQd): Failed to open stream: `' is not a valid mode for fopen in /home/runner/work/cakephp/cakephp/vendor/laminas/laminas-diactoros/src/Stream.php on line 330
PHP Warning:  fopen(/tmp/http_factory_tests_8EM9ce): Failed to open stream: `☠' is not a valid mode for fopen in /home/runner/work/cakephp/cakephp/vendor/laminas/laminas-diactoros/src/Stream.php on line 330
W
Warning: fopen(/tmp/http_factory_tests_8EM9ce): Failed to open stream: `☠' is not a valid mode for fopen in /home/runner/work/cakephp/cakephp/vendor/laminas/laminas-diactoros/src/Stream.php on line 330
```

**Comment by LordSimal** [comment]
The 1st is exactly the line I removed.
The 2nd is a problem in the embeded http_factory_tests which we can't adjust.

**Pull request merged**